charts/metering-ansible-operator/templates/crds/hive.crd.yaml

```diff
@@ -21,3 +21,216 @@ spec:
+              minLength: 1
+            tableProperties:
+              type: array
+              description: |
+                TableProperties is an array and allows you to tag the table definition with your
+                own metadata key/value pairs. Some predefined properties exist to control
+                behavior of the table as well.
+                Note: this is an optional field.
+                More info: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-listTableProperties
+            external:
+              type: boolean
+              description: |
+                External controls whether an external table is created, instead of a managed table.
+                If external is set to true, this causes Hive to point to an existing location,
+                as specified by the "location" field.
+                Note: this is an optional field.
+            managePartitions:
+              type: boolean
+              description: |
+                ManagePartitions controls whether the reporting-operator needs to check
+                if the table partitions match the partitions listed in the "partitions" field.
+                Note: this is an optional field.
+            partitions:
+              type: array
+              description: |
+                A list of partitions that this Hive table should contain.
+                Note: this is an optional field.
+              nullable: true
+              items:
+                type: object
+                required:
+                - partitionSpec
+                - location
+                properties:
+                  partitionSpec:
+                    type: array
+                    description: |
+                      PartitionSpec is a map containing string keys and values, where each key
+                      is expected to be the name of a partition column, and the value is the
+                      value of the partition column.
+                  location:
+                    type: string
+                    description: |
+                      Location specifies where the data for this partition is stored.
+                      This should be a sub-directory of the "location" field.
+                    minLength: 1
+                    format: uri
+          oneOf:
+          - required:
+            - numBuckets
+            - clusteredBy
+          - required:
+            - external
+            - location
+            properties:
+              external:
+                enum:
+                - true
+          - required:
+            - managePartitions
+            - partitions
+            properties:
+              managePartitions:
+                enum:
+                - true
+          - allOf:
+            - not:
+                required:
+                - numBuckets
+            - not:
+                required:
+                - clusteredBy
+            properties:
```

Comment:
Note: it's redundant to specify something like:
```yaml
properties:
  partitions:
    type: null
```
as we already attributed the `nullable: true` to the `partitions` array property.

Reply:
See: https://github.com/OAI/OpenAPI-Specification/issues/1900